packages/agent/src/actor.ts

```diff
@@ -530,25 +530,41 @@ function _createActorMethod(
         arg,
         effectiveCanisterId: ecid,
       });
-
-      requestId;
-      response;
-      requestDetails;
-
-      if (!response.ok || response.body /* IC-1462 */) {
-        throw new UpdateCallRejectedError(cid, methodName, requestId, response);
+      let reply: ArrayBuffer | undefined;
+      let certificate: Certificate | undefined;
+      if (response.body && response.body.certificate) {
+        const cert = response.body.certificate;
+        certificate = await Certificate.create({
+          certificate: bufFromBufLike(cert),
+          rootKey: agent.rootKey as ArrayBuffer,
+          canisterId: Principal.from(canisterId),
+          blsVerify,
+        });
+        const path = [new TextEncoder().encode('request_status'), requestId];
+        const status = new TextDecoder().decode(
+          lookupResultToBuffer(certificate.lookup([...path, 'status'])),
+        );
+
+        switch (status) {
+          case 'replied':
+            reply = lookupResultToBuffer(certificate.lookup([...path, 'reply']));
+            break;
+          case 'rejected':
+            throw new UpdateCallRejectedError(cid, methodName, requestId, response);
+          case 'accepted':
```

Comment:
`accepted` is not a valid[ request status](https://internetcomputer.org/docs/current/references/ic-interface-spec#state-tree-request-status) and is not part of the state tree. 

From v3 implementation, the request status could only be either `replied` or `rejected`